### PR TITLE
Switched to arrow functions

### PR DESCRIPTION
**Pull request by def14nt**
This PR changes all the `function ...` statements to arrow functions, and I also inserted brackets around all arrow function parameter names for more consistent styling.

### CODE REVIEW

**Comment by def14nt** [comment]
On a side note, I'm also looking into better middleware support.

**Comment by def14nt** [comment]
Sorry, this is why I avoid using Atom.

**Comment by def14nt** [comment]
oops

File: esm/html.js [hunk 1]
```diff
@@ -26,10 +26,8 @@ export const html = (query, ...args) => {
   return element;
 };
 
-html.extend = function (query, ...args) {
-  const clone = memoizeHTML(query);
-
-  return html.bind(this, clone, ...args);
+html.extend = (query, ...args) => {
+  return html.bind(this, memoizeHTML(query), ...args);
```
Comment (pakastin) [inline esm/html.js]: `=>` messes up `this`, I’m not sure if that’s wise here.. 🤔

File: esm/svg.js [hunk 2]
```diff
@@ -28,10 +28,8 @@ export const svg = (query, ...args) => {
   return element;
 };
 
-svg.extend = function (query) {
-  const clone = memoizeSVG(query);
-
-  return svg.bind(this, clone);
+svg.extend = (query) => {
+  return svg.bind(this, memoizeSVG(query));
```
Comment (pakastin) [inline esm/svg.js]: Same thing here: `=>` and `this` 🧐

File: esm/html.js (hunk 1 above)
Comment (def14nt) [reply esm/html.js]: Oh sorry, adjusting now. Was rushing through it, really think VSCode should add warnings for edge cases like this.

**Comment by def14nt** [comment]
Pushing to the remote now. Changes should be applied on your end. :smile:

**Comment by talmobi** [comment]
@def14nt "This PR changes all the function ... statements to arrow functions"... but why?

**Comment by pakastin** [comment]
Yeah, I also don't see the reason for this – so closing.

**Comment by def14nt** [comment]
@pakastin ES6?

**Comment by talmobi** [comment]
@def14nt Arrow functions aren't a replacement for function expressions or the like. For one they are anonymous (and thus harder to debug) and cannot self-reference for recursion or event handling like a named function would. Even though you can still achieve these things with variables, but then you have to take outer scope in mind (named functions can self-reference without polluting scope).

**Comment by pakastin** [comment]
Agreed, thanks @talmobi

**Comment by def14nt** [comment]
@talmobi Polluting scopes? `const a = () => {}` is equal to `function a() {}`.

**Comment by talmobi** [comment]
@def14nt that isn't a named function, however. I was justing highlighting that arrow functions aren't a defacto replacement for regular functions.

e.g.

```
const a = function giraffe ( val ) {
   if ( val < 10 ) return giraffe( val + 1 ) // self-reference
   return val
}
console.log( giraffe ) // undefined -- self-reference did not pollute outer scope.
```

**Comment by talmobi** [comment]
@def14nt here's another less contrived example:

```
;(function fun () { // function expression (IIFE)
  console.log( 'tick' )
  setTimeout( fun, 1000 )
})()

console.log( typeof fun ) // undefined
```

It's not like you can't achieve similar things with arrow functions but there's no reason to force everything to be arrow functions, especially if they make things less readable.

**Comment by pakastin** [comment]
If there’s a single reason to switch to arrow functions I’d do it, but currently there’s none.

In the worst case they mess up with context and even performance, so that’s a no from me.

**Comment by pakastin** [comment]
It’s a different thing to write a library than an application.

**Comment by def14nt** [comment]
> In the worst case they mess up with context and even performance, so that’s a no from me.

@pakastin Please explain how switching to a different syntax would/should affect performance in any way.

> It’s a different thing to write a library than an application.

What do you mean by this?

**Comment by def14nt** [comment]
@talmobi Let me show you how you would do that with an arrow function.

```js
const a = ( val ) => {
   if ( val < 10 ) return a( val + 1 ) // self-reference
   return val
}
console.log( giraffe ) // undefined -- self-reference did not pollute outer scope.
```

There we go. I fixed your pollution issue. I don't think you know what you're talking about.

**Comment by pakastin** [comment]
I do use arrow functions a lot when I need binded scopes or when it doesn’t matter they’re anonymous.

Using them everywhere is not wise.

**Comment by talmobi** [comment]
> @talmobi Let me show you how you would do that with an arrow function.
> 
> ```js
> const a = ( val ) => {
>    if ( val < 10 ) return a( val + 1 ) // self-reference
>    return val
> }
> console.log( giraffe ) // undefined -- self-reference did not pollute outer scope.
> ```
> 
> There we go. I fixed your pollution issue. I don't think you know what you're talking about.

@def14nt 

Well, not quite, you just have to learn to read. They're not identical even though, like I said before, you can achieve similar things with arrow functions.

Read up on the differences between function declarations, expression, named/anonymous functions and arrow functions etc. Top google SO result should do the trick.

For example you could do this with a named function:

```
el.addEventListener( 'click', function bar () { console.log( 'bar' ); el.removeEventListener( 'click', bar ) } )
```

With arrow functions you would either have to wrap inside an IIFE to hold the variable reference in order to not pollute scope that you would be creating to hold the arrow function (or without IIFE and pollute the current scope) or rely on the `.once`  handler (but that would leave the function without a name and become harder to debug). Possible? yes. Necessary? No.

But all of this is really off-topic, point is there is no de-facto benefit or need for forcing everything to be arrow functions.

Here are some links for you to read up on:
https://stackoverflow.com/questions/27977525/how-do-i-write-a-named-arrow-function-in-es2015
https://stackoverflow.com/questions/34361379/arrow-function-vs-function-declaration-expressions-are-they-equivalent-exch
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

**Comment by def14nt** [comment]
@talmobi Well it's your library, so it's up to you. I can see why arrow functions might not be preferable in some cases.

Instead of trying to correct me, I think you should thank me for my efforts. I didn't have to even submit this.

**Comment by def14nt** [comment]
@pakastin You've seen this *closed* pull request, yet you've completely ignored an *open* one for 16 days. The only open pull request is open to discussion, yet there hasn't been any. :thinking:

**Comment by pakastin** [comment]
Sorry, yeah I’ve been really busy

**Comment by def14nt** [comment]
@pakastin My advice to you would be to focus your efforts on constructive content, and avoid commenting on closed Pull Requests.

**Comment by pakastin** [comment]
Easy to do with mobile because no need to study code

**Comment by talmobi** [comment]
> @talmobi Well it's your library, so it's up to you. I can see why arrow functions might not be preferable in some cases.
> 

It's not my library. This isn't a tyranny, we're having a discussion about it right here, right now.

> Instead of trying to correct me, I think you should thank me for my efforts. I didn't have to even submit this.

> @pakastin My advice to you would be to focus your efforts on constructive content, and avoid commenting on closed Pull Requests.

This kind of attitude is definitely a good way to make sure others will avoid correcting, helping, and communicating with you in the future.

**Comment by def14nt** [comment]
@talmobi Tyranny?

I was giving advice. Is there something wrong? I have **suggested** to @pakastin he should stop commenting on **old Pull Requests** when I have made another that has been abandoned for 12 days.

You may think my attitude is unacceptable, but having a Pull Request that you put quite a bit of time into **ignored** because the owner would rather back up one of the contributors *does* annoy you after a while. I have taken the time to contribute to this codebase, so please treat my efforts with respect and not ignorance.

Stop typing your argumentative drivel and think about the effect that this will/has had on contributors. If you would rather I not give my advice to you, *please* tell me, because I'm trying to show you how this looks from the outside. I'm sorry you feel the correct me, while you completely miss the point.

Have a great day.
\- def14nt

**Comment by talmobi** [comment]
@def14nt 

You weren't giving advice, you were sarcastically pissing over his personal use of time. The two things aren't mutually exclusive.

You have not defended your position or given any good reasons as to why changing every function into an arrow function would be a good idea.

We have on the other hand pointed out and helped you understand why it may not be a good idea. And all you have done is piss all over our help while at the same time given no reasons of your own as to why retroactively (and by implication proactively) forcing everything to be an arrow function would be a good idea

You even had to fix an issue that came up in this very PR of you making every function an arrow function.

Just because you make a PR doesn't mean it's necessarily appropriate. We're all busy here and nobody is getting paid for this and we may not have the time to sit down for 15 mins to look over a PR in detail.

And your petty, sarcastic, and narcissistic attitude certainly isn't winning any points with me personally. I'm barely a contributor to this project with effectively no say whatsoever in its development. Don't confuse the status of this PR to some kind of conspiracy.

The only reason I don't like this PR is because you have given no good reason for anyone to like it other than implicitly (and unnecessarily) assuming that arrow functions are better in every way.

All this PR does is decrease readability, add confusion and increase the probability of bugs.

PR's are always welcome and it's a great thing. There's always an opportunity to learn, grow, and contribute to the project. Regardless if the PR is merged or not.

**Pull request closed without merging**